src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.cs

```diff
@@ -280,6 +286,35 @@ private static Glyph GetGlyph(SymbolKind kind, Accessibility? declaredAccessibil
             return result.ToImmutable();
         }
 
+        private static void AddNamesFromExistingOverloads(CSharpSyntaxContext context, SemanticModel semanticModel, ArrayBuilder<(string name, SymbolKind kind)> result, CancellationToken cancellationToken)
+        {
+            var namedType = semanticModel.GetEnclosingNamedType(context.Position, cancellationToken);
+            if (namedType is null)
+                return;
+
+            var parameterSyntax = context.LeftToken.GetAncestor(n => n.IsKind(SyntaxKind.Parameter)) as ParameterSyntax;
+            if (parameterSyntax is not { Type: { } parameterType, Parent: { Parent: MethodDeclarationSyntax method } })
+                return;
+
+            var overloads = namedType.GetMembers(method.Identifier.ValueText).WhereAsArray(m => m.Kind == SymbolKind.Method);
+            if (overloads.IsDefaultOrEmpty)
+                return;
+
+            var methodParameterType = semanticModel.GetTypeInfo(parameterType, cancellationToken).Type;
+            if (methodParameterType is null)
+                return;
+
+            foreach (var overload in overloads)
+            {
+                var methodSymbol = (IMethodSymbol)overload;
+                foreach (var overloadParameter in methodSymbol.Parameters)
+                {
+                    if (methodParameterType.Equals(overloadParameter.Type, SymbolEqualityComparer.Default))
+                        result.Add((overloadParameter.Name, SymbolKind.Parameter));
```

Comment:
do we need to dedupe this at all?  i can imagine many overloads will have the same name/parameter.  si that a problem?  or does a later place work that out.  thanks!

Reply:
hrmm... do we need to look at all parameters, or jsut parameters at the index we're at?  this feels a bit weird, but maybe it's nbd since we're using the type match to filter things down.

Reply:
> do we need to look at all parameters, or jsut parameters at the index we're at?

We need to look all. See the example in the linked issue, the parameter doesn't have the same index.

Reply:
> do we need to dedupe this at all? i can imagine many overloads will have the same name/parameter. si that a problem? or does a later place work that out. thanks!

It looks like there is some logic for this in:

https://github.com/dotnet/roslyn/blob/4ab21f1b9b6125c783531ab7a0d2479efd5fe306/src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs#L655

Reply:
gotcha.  i do feel like you can remove the deduping step.

Reply:
n/m.